### DISCUSSION TITLE
[Java] How is hierarchy in reports is created? Labels?

### DISCUSSION THREAD

**Discussion by alex-buyanov** [allure-framework/allure2#2312] [Questions & Support] [answered]
Preamble: I use Allure extensively in our CI/CD pipelines, so everyone on the team is accustomed to how reports look like. Recently, a new (proprietary) tool was introduced to the workflow. The tool itself is great, but it only generates reports in its custom JSON format. This JSON is easy to map onto Allure model with suites/test cases/steps/etc.

The question: how to create a custom "results processor" to which I can feed this JSON report and get an Allure report out?

I've found that I can use FileSystemResultsWriter instance to write TestResults. I was also able to figure out how to generate test cases using this TestResult class (and fixtures, and test containers, etc. etc.).

I cannot, however, understand how to create suites to make my report look nice. Could someone enlighten me on this please? :)

**Comment by alex-buyanov** [comment]
Scratch the original question. I was slopoking and didn't check how Allure creates its result files when I run "real" tests.

Is my understanding correct, that Allure uses labels on test resuts to create hierarchical view in the report?

Namely, `package`, `testClass`, `testMethod`, `parentSuite`, `suite`, and `subSuite` labels?

**Comment by baev** [comment, marked as answer]
You can find the information regarding the results format in our documentation: https://allurereport.org/docs/how-it-works-test-result-file/
Additional information on labels — https://github.com/orgs/allure-framework/discussions/2059#discussioncomment-6616266

**Reply by alex-buyanov** [reply]
Thank you very much!
